### ISSUE TITLE
Some research fails due to some particular characters as the `.`

### ISSUE THREAD

**Issue by silversonicaxel** [elamperti/OpenWebScrobbler#83]
**Describe the bug**
Some particular types of album/artist search are not working properly, maybe due to special characters as `.` or else.

**Steps to reproduce**
1. Go to album / artist search page https://openscrobbler.com/scrobble/album 
2. Type the name of this Icelandic artist `dj. flugvél og geimskip`
3. No albums or no artists are shown

**Expected behavior**
Albums and artist related to the search `dj. flugvél og geimskip` should appear

**Device**
  * [x] Desktop
  * [x] Mobile

**Platform (please complete the following information):**
 - Operating system (name and version): 
 - Browser (name and version):
 - Version (see footer of website):

**Additional context**
The network tab of the browser dev area shows that API call return correctly the results, so error is within the javascript code.
The search of ` flugvél og geimskip` returns results correctly. That's the reason of the assumption of special characters as cause of the bug.

**Comment by elamperti** [comment]
Thanks for this detailed report! I was able to find the source of this error: when the search query (or the artist or the album name) contained periods, the lodash `get()` was interpreting them as a separator between an object `dj` and a key ` flugvél og geimskip`. This was solved properly _escaping_ the query string by referencing it inside quotes, e.g.:

```
myObject.get(data, `foo['${query}'].bar`)
```

This can be seen more clearly in the commit fea8420, which solves this issue.

I also improved on how the artists were being referenced, as sometimes the searches would show empty results when the artist didn't have an `mbid`.
